portafly/src/dal/accounts/getDeveloperAccounts.ts

```diff
@@ -0,0 +1,17 @@
+import { useFetch } from 'react-async'
+import { IDeveloperAccount } from 'types'
+
+/**
+ * Get all Applications asynchronously
+ * @returns An object containing: { accounts, error, isPending }
```

Comment:
What would you say about normalizing the response of the DAL layer with as follows:

`payload`:  would be of the desired data type (I.e: IDeveloperAccount, IAccountSettings, etc) OR error type.
`error`: Optional that  **may** be set to boolean `true` if the fetch action presented an error
`isPending`: Boolean, self explanatory.

Reply:
I think I'd rather leave it like it is to be honest. I'm no sure we should mix network errors with http error codes.. We would have to check payload every time for an error to set `error = true`...

Reply:
Why would you think we would be mixing network errors with http codes? And we wouldn't be checking the payload, just the error boolean. I thing this way the types would be unified and  our implementation consistent. We would only need to have a custom fn encapsulating `fetch` that we will be passing only the endpoint, and it'll be returning `payload`, `error` and `isPending`. We also will be avoiding repetition of passing the same headers, most probably AuthN/AuthZ, etc. Give it a second thought and maybe we could explore the interface together